astropy/coordinates/baseframe.py

```diff
@@ -359,6 +361,64 @@ def _insert(self, obj, values, axis=0):
         return out
 
 
+class CoordinateSharedMethods:
```

Comment:
The problem is that while just letting `__getattr__` take care works for `masked` and `mask`, it does not work for `.get_mask`, since it takes a series of attributes of the class and not all `SkyCoord` attributes are available on the frame (this is explicitly tested with `pressure`), i.e., `self` is required to the `SkyCoord`, not its underlying frame.

p.s. Of course, having this shared method mixin will also help remove duplication between the two like for `.insert()`.

Reply:
p.p.s. A side benefit of having a mixin class is that, eventually, we can actually have sphinx render the relevant properties and methods (once `:inherited-members:` works properly).

Reply:
Yes, just to keep things together, and with a view of expanding later. But if this is something that will prevent this PR from getting into 7.0, I'm fine with removing this update, and just duplicating `SkyCoord.get_mask`...

Reply:
`CoordinateSharedMethods` looks like a private implementation detail that we should be able to change without notice, but it does implement methods that are clearly a part of the public API. I am very hesitant to introduce such classes to `coordinates` until the APE about public API has been accepted.

I would implement methods on `BaseCoordinateFrame` if `SkyCoord` can use them via `__getattr__()` because there are several methods that are implemented like that already. For `get_mask()` it might be best to avoid code duplication by introducing a private `_get_mask()` that both `SkyCoord.get_mask()` and `BaseCoordinateFrame.get_mask()` can call.

Reply:
Yes, fair point that really it is best not to mix new types of implementation in. Will make the suggested changes.